assets/js/modules/analytics-4/components/setup/SetupForm.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Analytics Setup form.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { SpinnerButton, ProgressBar } from 'googlesitekit-components';
+import {
+	FORM_SETUP,
+	EDIT_SCOPE,
+	ENHANCED_MEASUREMENT_ENABLED,
+	ENHANCED_MEASUREMENT_FORM,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import { MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import { CORE_LOCATION } from '../../../../googlesitekit/datastore/location/constants';
+import { isPermissionScopeError } from '../../../../util/errors';
+import SetupFormFields from './SetupFormFields';
+import StoreErrorNotices from '../../../../components/StoreErrorNotices';
+
+import useViewContext from '../../../../hooks/useViewContext';
+import { trackEvent } from '../../../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function SetupForm( { finishSetup } ) {
+	const hasEditScope = useSelect( ( select ) =>
+		select( CORE_USER ).hasScope( EDIT_SCOPE )
+	);
+	const autoSubmit = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'autoSubmit' )
+	);
+	const canSubmitChanges = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).canSubmitChanges()
+	);
+	const isSaving = useSelect(
+		( select ) =>
+			select( MODULES_ANALYTICS_4 ).isDoingSubmitChanges() ||
+			select( CORE_LOCATION ).isNavigating()
+	);
+	const viewContext = useViewContext();
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	const { submitChanges } = useDispatch( MODULES_ANALYTICS_4 );
+	// @TODO remove when analytics 4 is completely decoupled from analytics module.
+	const { submitChanges: submitChangesLegacy } =
+		useDispatch( MODULES_ANALYTICS );
```

Comment:
This is not necessary as this component will not be surfaced for the end user until the modules are completely decoupled (as explained above).

```suggestion
```